lib/std/build.zig

```diff
@@ -1265,40 +1203,68 @@ fn isLibCLibrary(name: []const u8) bool {
     return false;
 }
 
+/// A file that is generated by a build step.
+/// This struct is an interface that is meant to be used with `@fieldParentPtr` to implement the actual path logic.
+pub const GeneratedFile = struct {
+    /// The step that generates the file
+    step: *Step,
+
+    /// A function that returns the absolute path to the generated file.
+    getPathFn: fn (self: *const GeneratedFile) []const u8,
```

Comment:
Does this need to be an interface?  A boxed string (`struct { path: []const u8 = "" }`) which gets updated in the make step would be simpler.  You could check if the string is still empty when read and give a nice panic message also.

Reply:
I don't think this is simpler in the end, especially as you require additional allocations, pointer references and such. Right now, we have a single forward reference to manage: `*const GeneratedFile`

Or do you mean `GeneratedFile` will look like this:
```zig
struct {
    step: *Step,
    path: ?[]const u8 = null,
}
```
And the generator requires to update `.path` to a non-null variant?

Reply:
Yeah, exactly, that second one.  You don't need extra allocations since steps are never deleted, just have the referencing step point into a GeneratedFile struct inside the referenced step.

If it's a function, it can have side effects and do all sorts of things.  It's not clear what operations the calling code can tolerate and what might break.  But like this it's very clear -- the referencer will only read this value, and the referenced step must set it before that happens.

Assuming `""` is an invalid value for a path, using that as a sentinel instead of an optional will save eight bytes, which is why I went for that version.

Reply:
I'll rework the PR to use this case. It's definitly the superior solution here. Thanks for pointing that out!

Reply:
Changed the code to use this pattern instead